src/llm_utils.py

```diff
@@ -0,0 +1,176 @@
+
+import os
+import json
+from google.genai import types
+from google import genai
+from typing import List, Dict
+
+def analyze_code_changes(diff_content: str) -> List[Dict]:
+    """
+    Analyze code changes using OpenAI's GPT model
+    Returns a list of review comments
+    """
+    API_KEY = os.getenv('GEMINI_API_KEY')
+
+    if not API_KEY:
+        raise ValueError("GEMINI_API_KEY is not set. Please add it to your environment variables.")
+
+    # Prepare the prompt for the LLM
+    prompt = f"""
+    Analyze the following code changes and provide review comments using EXACTLY this format:
+
+    FILE: [file-path]
+    LINE: [line-number]
+    COMMENT: [your comment]
+    SUGGESTION: [optional suggested code]
+
+    ---
+    Example:
+    FILE: src/app.py
+    LINE: 42
+    COMMENT: Avoid magic numbers, consider using a constant
+    SUGGESTION: MAX_RETRIES = 3
+
+    ---
+
+    Make sure to insert '---' in between two review comments. IT IS A MUST!!!
+
+    Focus on:
+    - Code quality issues
+    - Security vulnerabilities
+    - Performance optimizations
+    - Style inconsistencies
+    - Possible bugs
+    - Missing documentation
+
+    Diff content:
+    {diff_content}
+    """
+
+    client = genai.Client(api_key=API_KEY)
+
+    #Get analysis from Gemini model
+    response = client.models.generate_content(
+        model="gemini-2.0-flash", contents=prompt
+    )   
+
+    print(response.text)
+    
+    # Parse and format the response
+    review_comments = parse_llm_response(response.text)
+    return review_comments
+
```

Comment:
The `comment_data` dictionary is initialized with `path` and `line` set to `None`. If these remain `None` after processing a comment block, the comment should be skipped. This check ensures that the extracted data is valid before creating a review comment.

```suggestion
The current check `if has_file and has_line and has_comment:` does not enforce checking `comment_data['path']` and `comment_data['line']` are not `None`. This can cause issues when keys exists, but values are `None` after parsing.
```